core/pom.xml

```diff
@@ -89,7 +84,6 @@
         <dependency>
             <groupId>com.google.guava</groupId>
             <artifactId>guava</artifactId>
-            <scope>test</scope>
```

Comment:
OK then, that's fine by me.
re: shade plugin, I disabled it back when JDK8 was very new, and the shade plugin could not handle the new class file format yet.  I assume it works fine now and just needs testing.

However, I would push to have two artifacts -- maybe one without any classifier, and one with a `shaded` classifier?  At least for my projects I find shading in dependencies totally counterproductive, since we already have Guava and friends present, and that way we can pull in bugfixes with dependency management.  But if people really want shaded jars we can provide those too?